supervisor/shared/external_flash/spi_flash.c

```diff
@@ -42,9 +42,12 @@ const external_flash_device* flash_device;
 uint32_t spi_flash_baudrate;
 
 // Enable the flash over SPI.
-static void flash_enable(void) {
-    while (!common_hal_busio_spi_try_lock(&supervisor_flash_spi_bus)) {}
-    common_hal_digitalio_digitalinout_set_value(&cs_pin, false);
+static bool flash_enable(void) {
+    if (common_hal_busio_spi_try_lock(&supervisor_flash_spi_bus)) {
```

Comment:
Is it worth retrying X amount of times even? I know `adafruit_bus_device` had a routine to catch background tasks and keep trying for the lock, I'm not familiar enough to say for sure this needs it just a thought.

Reply:
Could it be that instead of the busy-wait in https://github.com/adafruit/Adafruit_CircuitPython_BusDevice/blob/master/adafruit_bus_device/spi_device.py#L73, it's better to throw an exception when the SPI is locked? Then the user would have control over how many times, if at all, to retry.

Reply:
Looping over `common_hal_busio_spi_try_lock()` in C will not succeed if it doesn't work the first time: If foreground Python code has locked the bus, it has to be able to proceed in order to **un**lock the bus.  I don't think we ever un/lock the bus from an interrupt context.

Reply:
@jepler doesn't the same apply to `adafruit_bus_device`? Or may `run_background_tasks()` lock SPI in one tick, and unlock it in another?